applications/HDF5Application/custom_io/hdf5_file.cpp

```diff
@@ -253,6 +298,74 @@ void File::AddPath(const std::string& rPath)
     }
 }
 
+template <class TScalar>
+void File::WriteAttribute(const std::string& rObjectPath, const std::string& rName, TScalar Value)
+{
+    KRATOS_TRY;
+    BuiltinTimer timer;
+    hid_t type_id, space_id, attr_id;
+
+    type_id = Internals::GetScalarDataType<TScalar>();
+    space_id = H5Screate(H5S_SCALAR);
+    KRATOS_ERROR_IF(space_id < 0) << "H5Screate failed." << std::endl;
+    attr_id = H5Acreate_by_name(m_file_id, rObjectPath.c_str(), rName.c_str(), type_id,
+                                space_id, H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT);
+    KRATOS_ERROR_IF(attr_id < 0) << "H5Acreate_by_name failed." << std::endl;
+    KRATOS_ERROR_IF(H5Awrite(attr_id, type_id, &Value) < 0) << "H5Awrite failed." << std::endl;
+    KRATOS_ERROR_IF(H5Sclose(space_id) < 0) << "H5Sclose failed." << std::endl;
+    KRATOS_ERROR_IF(H5Aclose(attr_id) < 0) << "H5Aclose failed." << std::endl;
+    if (GetEchoLevel() == 2 && GetPID() == 0)
+        std::cout << "Write time \"" << rObjectPath << '/' << rName << "\": " << timer.ElapsedSeconds() << std::endl;
```

Comment:
```suggestion
        KRATOS_INFO_IF("Write time \"" << rObjectPath << '/' << rName << "\":, GetEchoLevel() == 2) << timer.ElapsedSeconds() << std::endl;
```

sth like this

by using the logger it will automatically only print on one rank (after #4018 which is pretty much ready to be merged)

Reply:
I agree with the suggestion to use the logger, but I would ask you to use a single consistent label (such as `HFD5IO` or whatever you consider appropriate), since it helps filter messages by source.